pageserver/src/http/routes.rs

```diff
@@ -79,13 +73,27 @@ fn get_config(request: &Request<Body>) -> &'static PageServerConf {
     get_state(request).conf
 }
 
-// Helper functions to construct a LocalTimelineInfo struct for a timeline
-
-fn local_timeline_info_from_timeline(
-    timeline: &Arc<Timeline>,
+async fn build_timeline_info(
+    timeline: &Timeline,
     include_non_incremental_logical_size: bool,
     include_non_incremental_physical_size: bool,
-) -> anyhow::Result<LocalTimelineInfo> {
+) -> anyhow::Result<TimelineInfo> {
+    let mut info = build_timeline_info_guts(timeline)?;
+    if include_non_incremental_logical_size {
```

Comment:
Both operations might take long time, makes sense to run them both with [`try_join`](https://docs.rs/tokio/latest/tokio/macro.try_join.html) so that they _start_ calculating more or less at the same time.

Right now we never start the 2nd task before the first one completes, which presumably takes more time if both calculations are queried.